test/PositionManager.ts

```diff
@@ -123,18 +171,167 @@ describe("GammaPoolFactory", function () {
+            expect(args.lpTokensPaid.toNumber()).to.equal(25);
+            expect(args.amountsLen.toNumber()).to.equal(9);
+        });
+
+        it("#increaseCollateral should return tokenId and length of tokens held", async function () {
+            await tokenA.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            await tokenB.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
+            const AddRemoveCollateralParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: 1,
+                amounts: [100,10],
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+            
+            const res = await (await posMgr.increaseCollateral(AddRemoveCollateralParams)).wait();
+            
+            const { args } = res.events[2];
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.tokenId.toNumber()).to.equal(1);
+            expect(args.tokensHeldLen.toNumber()).to.equal(6);
+        });
 
+        it("#decreaseCollateral should return tokenId and length of tokens held", async function () {
+            const AddRemoveCollateralParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: 1,
+                amounts: [100,10],
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+            
+            const res = await (await posMgr.decreaseCollateral(AddRemoveCollateralParams)).wait();
+            
+            const { args } = res.events[0]
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.tokenId.toNumber()).to.equal(1);
+            expect(args.tokensHeldLen.toNumber()).to.equal(7);
+        });
+
+        it("#rebalanceCollateral should return tokenId and length of tokens held", async function () {            
+            const RebalanceCollateralParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: 1,
+                deltas: [4, 2],
+                liquidity: 1,
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+            
+            const res = await (await posMgr.rebalanceCollateral(RebalanceCollateralParams)).wait();
+            
+            const { args } = res.events[0]
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.tokenId.toNumber()).to.equal(1);
+            expect(args.tokensHeldLen.toNumber()).to.equal(10);
+        });
+
+        it("#rebalanceCollateralWithLiquidity should return tokenId and length of tokens held", async function () {
+            const RebalanceCollateralParams = {
+                cfmm: cfmm.address,
+                protocol: 1,
+                tokenId: 1,
```

Comment:
use tokenId from beforeEach here, don't just use 1. So it should be "tokenId: tokenId,"